stdlib/public/Concurrency/Task.swift

```diff
@@ -186,6 +186,24 @@ extension Task: Equatable {
   }
 }
 
+@available(SwiftStdlib 5.9, *)
+extension Task where Failure == Never {
+    @_spi(MainActorUtilities)
+    @MainActor
+    static func taskStartingOnMainActor(
+        priority: TaskPriority? = nil,
+        _ work: @escaping @MainActor() async -> Success
+    ) -> Task<Success, Never> {
+        let flags = taskCreateFlags(priority: priority, isChildTask: false,
+                                    copyTaskLocals: false, inheritContext: false,
```

Comment:
Hmm. I think this probably should inherit context, because it's meant for main-thread work?

Reply:
Yeah, I think this needs the `@_unsafeInheritExecutor` treatment